routes/carts.js

```diff
@@ -0,0 +1,67 @@
+const express = require("express");
+const router = express.Router();
+const db = require("../database/index");
+
+router.delete("/:user_id/:product_id", (req, res) => {
+  let user_id = req.params.user_id;
+  let prod_id = req.params.product_id;
+  let errorMessage;
+  let arr = [];
+  db.raw(`SELECT * FROM "Carts" WHERE user_id = ?`, [user_id])
+    .then(results => {
+      if (results.rows.length === 0) {
+        errorMessage = "Cannot find user";
+        throw err;
+      }
+
+      results.rows.forEach(e => {
+        arr.push(e.products_id);
+      });
+
+      if (!arr.includes(parseInt(prod_id))) {
+        errorMessage = "Cannot find product";
+        throw err;
+      } else {
+        db.raw(
```

Comment:
Always return promises and don't nest them. if you returned this promise, the result of the inner promise will go the the .then of the outer promise.